src/main/kotlin/com/example/ipwhitelist/repository/AppRepository.kt

```diff
@@ -0,0 +1,223 @@
+import com.example.ipwhitelist.model.dynamodb.ApplicationDetails
+import com.example.ipwhitelist.model.dynamodb.ApplicationUser
+import org.springframework.stereotype.Repository
+import software.amazon.awssdk.enhanced.dynamodb.*
+import software.amazon.awssdk.enhanced.dynamodb.model.*
+import software.amazon.awssdk.enhanced.dynamodb.model.QueryConditional.sortBeginsWith
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue
+import java.util.*
+import kotlin.collections.ArrayList
+
+@Repository
+class AppRepository(
+    val dynamoDbEnhancedClient: DynamoDbEnhancedClient
+) {
+    fun <T : Application> save(application: T): T {
+        val applicationTable = getTable(application.javaClass)
+        applicationTable.putItem(application)
+        return application
+    }
+
+    fun getAllApps(): List<ApplicationDetails> {
+        val appTable = getTable(ApplicationDetails::class.java)
+        val items: MutableList<ApplicationDetails> = ArrayList()
+
+        val scanEnhancedRequest = ScanEnhancedRequest.builder()
+            .filterExpression(
+                Expression.builder()
+                    .expression("begins_with(objectId, :prefix)")
+                    .expressionValues(Collections.singletonMap(
+                        ":prefix", AttributeValue.builder()
+                            .s(AppTableKeyPrefix.APP.prefix).build()))
+                    .build()
+            )
+            .build()
+
+        val pageIterable: PageIterable<ApplicationDetails> = appTable.scan(scanEnhancedRequest)
+
+        for (page: Page<ApplicationDetails> in pageIterable) {
+            items.addAll(page.items())
+        }
+
+        return items
+    }
+
+    fun getAppsByUserId(userId: String) : List<EnhancedAppResponse> {
+
+        val scanEnhancedRequest = ScanEnhancedRequest.builder()
+            .filterExpression(
+                Expression.builder()
+                    .expression("objectId = :objId")
+                    .expressionValues(
+                        Collections.singletonMap(
+                            ":objId", AttributeValue.builder()
+                                .s(userId).build()
+                        )
+                    )
+                    .build()
+            )
+            .build()
+
+        val appUsersTable = getTable(ApplicationUser::class.java)
+        val pageIterable: PageIterable<ApplicationUser> = appUsersTable.scan(scanEnhancedRequest)
+
+        val authorizedApps: MutableList<ApplicationUser> = ArrayList()
+        for (page: Page<ApplicationUser> in pageIterable) {
+            authorizedApps.addAll(page.items())
+        }
+
+        // using the authorized appIds, fetch the corresponding app details from the ApplicationDetails table
+        val appDetailsTable = getTable(ApplicationDetails::class.java)
+        val enhancedAppResponses: MutableList<EnhancedAppResponse> = ArrayList()
+
+        authorizedApps.forEach { app ->
```

Comment:
For batch requests, there is a ReadBatch class that can send the requests in one batch instead of sending multiple query requests